rogue-spindle/src/main/scala/com/foursquare/rogue/spindle/SpindleDBDecoderFactory.scala

```diff
@@ -0,0 +1,77 @@
+// Copyright 2015 Foursquare Labs Inc. All Rights Reserved.
+
+package com.foursquare.rogue.spindle
+
+import com.foursquare.spindle.{UntypedMetaRecord, UntypedRecord}
+import com.mongodb.{BasicDBObject, DBCursor, DBCallback, DBCollection, DBDecoder, DBDecoderFactory, DBObject}
+import java.io.InputStream
+import org.bson.{BasicBSONDecoder, BSONCallback, BSONObject}
+
+case class SpindleDBDecoderFactory(
+  meta: UntypedMetaRecord,
+  recordReader: (UntypedMetaRecord, InputStream) => SpindleDBObject
+) extends DBDecoderFactory {
+  def create(): DBDecoder = new SpindleDBDecoder(meta, recordReader)
+}
+
+case class SpindleDBDecoder(
+  meta: UntypedMetaRecord,
+  recordReader: (UntypedMetaRecord, InputStream) => SpindleDBObject
+) extends DBDecoder {
+  def decode(is: InputStream, collection: DBCollection): DBObject = {
+    recordReader(meta, is)
+  }
+
+  def getDBCallback(collection: DBCollection): DBCallback = {
+    throw new UnsupportedOperationException()
```

Comment:
I looked through the mongo-java-driver codebase and getDBCallback is never used. decode(bytes) is used for testing.